src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -170,9 +172,58 @@ public int GetCount(bool onlyIfCheap)
             {
                 return _source.Length;
             }
+
+            public IPartition<TResult> Skip(int count)
+            {
+                if (count == 0) return (IPartition<TResult>)Clone();
+                if (count >= _source.Length) return new EmptyPartition<TResult>();
+                return new SelectIPartitionIterator<TSource, TResult>(new ListPartition<TSource>(_source, count, int.MaxValue), _selector);
+            }
+
+            public IPartition<TResult> Take(int count)
+            {
+                return count >= _source.Length ? (IPartition<TResult>)Clone() : new SelectIPartitionIterator<TSource, TResult>(new ListPartition<TSource>(_source, 0, count - 1), _selector);
+            }
```

Comment:
It could, but that arguably becomes less natural if one is thinking in terms of pages of output. It's also more likely if one uses the query syntax to put those methods not supported by it outside, and more likely also if one has more that one possible source of strings (or whatever the element type is).

Also the developer who takes skip-first approach doesn't suffer because of this, It seems from a quick check they very slightly gain from SelectIPartition though I wouldn't put too much stock in that one, it's too close to mean much. They certainly do gain if they `ToArray()` or `ToList()` the result though.

I do agree that well-written queries should take priority over poor, but I'm also hoping that this helps to "light up" a lot of the other optimisations that have been done, and does so without the cost of added checks, since the interfaces involved are already looked for. (There is one added check in `Select` to catch ordered enumerables, which is a much more reasonable case since the benefit accrues whichever order one does the skipping and taking).